src/background.ts

```diff
@@ -151,14 +166,330 @@ const onStartUp = async () => {
   checkIfProtected(store.getState());
 
   browser.tabs.onUpdated.addListener(TabEvents.onDomainChange);
+  browser.tabs.onUpdated.addListener(TabEvents.onTabDiscarded);
   browser.tabs.onUpdated.addListener(TabEvents.onTabUpdate);
   browser.tabs.onRemoved.addListener(TabEvents.onDomainChangeRemove);
-  browser.tabs.onRemoved.addListener(TabEvents.cleanFromFromTabEvents);
+  browser.tabs.onRemoved.addListener(TabEvents.cleanFromTabEvents);
 
   // This should update the cookie badge count when cookies are changed.
   browser.cookies.onChanged.addListener(CookieEvents.onCookieChanged);
+
+  if (browser.contextMenus) {
+    ContextMenuEvents.menuInit(store.getState());
+    if (!browser.contextMenus.onClicked.hasListener(onContextMenuClicked)) {
+      browser.contextMenus.onClicked.addListener(onContextMenuClicked);
+    }
+  }
+  browser.browserAction.setTitle({ title: `${mf.name} ${mf.version} [READY] (0)` });
+  if (currentSettings.debugMode.value) {
+    cadLog({
+      msg: `background.onStartUp is complete.`,
+      type: "info"
+    });
+  }
 };
 
+async function onContextMenuClicked(
+  info: browser.contextMenus.OnClickData,
+  tab: browser.tabs.Tab
+) {
+  const debug = getSetting(store.getState(), 'debugMode') as boolean;
+  const contextualIdentities = getSetting(store.getState(), 'contextualIdentities') as boolean;
+  if (debug) {
```

Comment:
Instead of if (debug) statements everywhere, might I suggest just calling 

```
    cadLog({
      msg: `background.onContextMenuClicked:  Data received`,
      x: {info, tab},
    });
```
And then inside the cadLog function, it would have the `if (debug)` statement. Doing this should reduce the amount of test cases as well since there's no more condition.

Reply:
So what you're saying is to pass another variable into cadLog, call it debug, and then in each of the calling functions pass in ```getSetting(store.getState(), 'debugMode') as boolean```?

Reply:
Heads up - when I'm done working on this, more files will be changed.

Normally I'd say if a function doesn't need to be called then use `if`, but if you want to minimize test cases then we'll do it that way then.

Reply:
> So what you're saying is to pass another variable into cadLog, call it debug, and then in each of the calling functions pass in getSetting(store.getState(), 'debugMode') as boolean?

Yeah that would work.

Eventually though we may want to do some dependency injection (i.e. make cadLog into a singleton class function that has the state of the extension) so we don't have to pass in  `getSetting(store.getState(), 'debugMode')`. But your solution will work for the short term.

Reply:
I'd haven't thought about moving cadLog from Libs to a separate class...I had thought to keep the multiple use functions into Libs.

Mind you this was only an initial debug log - I had intend for it to either show up on a new settings page for easier debugging/exporting.

Also in the future this debug should have levels as there is so much information being sent through.

Reply:
The only other thing is that there are some cadLog calls that still outputs regardless of debug mode, so if we're moving it into a separate class that needs to be taken into consideration.

Reply:
Should be resolved through 2b5e3b2, at least through adding in the parameter for now.